Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs

```diff
@@ -320,10 +320,7 @@ void BatchUpdate(Action update)
 			},
 					(_) =>
 					{
-						if (_batchUpdating.CurrentCount > 0)
-						{
-							_batchUpdating.Release();
-						}
+						_batchUpdating.Release();
```

Comment:
@hartez `CurrentCount` is 

> The number of remaining threads that can enter the semaphore.

so checking here if CurrentCount > 0 probably isn't correct.

Also `_batchUpdating.Wait();` is going to block the thread so I don't feel like that is right either

Would this work?
```C#
async void BatchUpdate(Action update)
		{
			await _batchUpdating.WaitAsync();
			_collectionView.PerformBatchUpdates(() =>
			{
				update();
			},
			(_) =>
			{
				_batchUpdating.Release();
			});
		}
```

Reply:
Yeah, what's in here right now is wrong. I fixed this for ObservableGroupedSource, but forgot to apply the same fix here: https://github.com/xamarin/Xamarin.Forms/pull/9911/commits/220715c11b9543b6b2b7d98ac9acefddc54e1db8

Reply:
> Also _batchUpdating.Wait(); is going to block the thread so I don't feel like that is right either

It won't block the thread because the semaphore is available. Since `CurrentCount` is greater than 0, `Wait()` decrements the count and returns immediately. 

```
{
	_batchUpdating.Release();
});
```

We need the check for `CurrentCount == 0` here. If multiple calls to `PerformBatchUpdates` are batched together, blindly calling `Release()` could result in an `SemaphoreFullException` (https://docs.microsoft.com/en-us/dotnet/api/system.threading.semaphoreslim.release?view=netcore-3.1#System_Threading_SemaphoreSlim_Release)